bioagents/tests/msa_test.py

```diff
@@ -64,6 +65,110 @@ def test_no_activity_given():
     _check_failure(msg, 'getting no activity type', 'UNKNOWN_ACTION')
 
 
+class _TestMsaGeneralLookup(_IntegrationTest):
+    def __init__(self, *args, **kwargs):
+        super(_TestMsaGeneralLookup, self).__init__(msa_module.MSA_Module)
+
+    def _get_content(self, task, **contents):
+        content = KQMLList(task)
+        for key, value in contents.items():
+            content.set(key, value)
+        msg = get_request(content)
+        return msg, content
+
+    def _get_tells(self, logs):
+        return [msg for msg in logs if msg.head() == 'tell']
+
+    def _get_provenance_tells(self, logs):
+        provenance_tells = []
+        for msg in logs:
+            if msg.head() == 'tell' and msg.get('content'):
+                content = msg.get('content')
+                if content.head() == 'add-provenance':
+                    html = content.gets('html')
+                    provenance_tells.append(html.splitlines()[0])
+        return provenance_tells
+
+    def _check_find_response(self, output):
+        assert output.head() == 'SUCCESS', str(output)
+        logs = self.get_output_log()
+        prov_tells = self._get_provenance_tells(logs)
+        assert len(prov_tells) == 1, prov_tells
+
+
+class TestMSATypeAndTarget(_TestMsaGeneralLookup):
+    def create_type_and_target(self):
+        return self._get_content('FIND-RELATIONS-FROM-LITERATURE',
+                                 source=ekb_from_text('None'),
```

Comment:
Is this ekb_from_text('None') intentional? That would imply a situation in which the BA literally sends the EKB of the string "None" to the bioagent which wouldn't happen in practice.

Reply:
This absolutely happens. It is how we are able to answer general questions for causal information with just one (or two, if you include the confirmation variation) capabilities. Anything which is not given, but is grammatically implied as missing (_What_ phosphorylates BRAF?) will still be captured, but turned to None.

Reply:
But the BA certainly doesn't take the string "None", send it for EKB conversion to the EKB Agent and then send you an XML of it (if it does anything like that that's a bug on that end). I imagine if one of the arguments is missing, it would/should send ```:source ()``` or omit the :source argument from the request.

Reply:
That would require encoding a few different code paths. What we get is a non-protein "object" which the ekb renders into something which can be identified with None. Like I said, that how I wrote the BA capability to accommodate the different use cases.